climpred/smoothing.py

```diff
@@ -0,0 +1,230 @@
+    ds,
+    d_lon_lat_dict={'lon': 5, 'lat': 5},
+    method='bilinear',
+    periodic=False,
+    filename=None,
+    reuse_weights=True,
+):
+    """
+    Quick regridding
+
+    Parameters
+    ----------
+    ds : xarray DataSet
+        Contain input and output grid coordinates. Look for variables
+        ``lon``, ``lat``, and optionally ``lon_b``, ``lat_b`` for
+        conservative method.
+         Shape can be 1D (Nlon,) and (Nlat,) for rectilinear grids,
+        or 2D (Ny, Nx) for general curvilinear grids.
+        Shape of bounds should be (N+1,) or (Ny+1, Nx+1).
+     d_lon_lat_dict : dict, optional
+        Longitude/Latitude step size, i.e. grid resolution; if not provided,
+        Longitude will equal 1 and Latitude will equal Longitude
+     method : str
+        Regridding method. Options are
+        - 'bilinear'
+        - 'conservative', **need grid corner information**
+        - 'patch'
+        - 'nearest_s2d'
+        - 'nearest_d2s'
+     periodic : bool, optional
+        Periodic in longitude? Default to False.
+        Only useful for global grids with non-conservative regridding.
+        Will be forced to False for conservative regridding.
+     filename : str, optional
+        Name for the weight file. The default naming scheme is::
+             {method}_{Ny_in}x{Nx_in}_{Ny_out}x{Nx_out}.nc
+         e.g. bilinear_400x600_300x400.nc
+     reuse_weights : bool, optional
+        Whether to read existing weight file to save computing time.
+        False by default (i.e. re-compute, not reuse).
+
+    Returns
+    -------
+    ds : xarray DataSet with coordinate values or DataArray
+    """
+
+    if xe is None:
+        raise ImportError(
+            'xesmf is not installed; see https://xesmf.readthedocs.io/en/latest/installation.html'
+        )
+
+    def _regrid_it(da, d_lon, d_lat, **kwargs):
+        """
+        Global 2D rectilinear grid centers and bounds
+         Parameters
+        ----------
+        da : xarray DataArray
+            Contain input and output grid coordinates. Look for variables
+            ``lon``, ``lat``, and optionally ``lon_b``, ``lat_b`` for
+            conservative method.
+             Shape can be 1D (Nlon,) and (Nlat,) for rectilinear grids,
+            or 2D (Ny, Nx) for general curvilinear grids.
+            Shape of bounds should be (N+1,) or (Ny+1, Nx+1).
+         d_lon : float
+            Longitude step size, i.e. grid resolution
+         d_lat : float
+            Latitude step size, i.e. grid resolution
+         Returns
+        -------
+        da : xarray DataArray with coordinate values
+        """
+
+        def warn_lon_lat_dne(da):
+            if 'lat' in ds.coords and 'lon' in ds.coords:
+                return da
+            # elif 'lat_b' in ds.coords and 'lon_b' in ds.coords:
+            #    da = da.rename({'lat_b': 'lat', 'lon_b': 'lon'})
+            #    return da
+            elif 'TLAT' in ds.coords and 'TLONG' in ds.coords:
```

Comment:
I would add a comment here mentioning that this is for the POP grid (e.g. CESM) support. We can add explicit support for other models as they come along. Or could look through the CMIP6 archive to see what we should be supporting.

Reply:
the value error afterwards asks for renaming if it doesnt work